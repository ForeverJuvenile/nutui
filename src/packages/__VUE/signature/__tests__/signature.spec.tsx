import { mount } from '@vue/test-utils';
import Signature from '../index.vue';
import { createCanvas } from '@napi-rs/canvas';

test('props custom-class', async () => {
  const wrapper = mount(() => {
    return <Signature customClass="test" line-lineWidth={5} strokeStyle="#333"></Signature>;
  });
  const signatureWrapper = wrapper.find('.nut-signature');
  expect(signatureWrapper.classes()).toContain('test');

  const canvas = createCanvas(400, 400);
  const ctx = canvas.getContext('2d');
  ctx.lineWidth = 5;
  ctx.strokeStyle = '#333';
  ctx.beginPath();
  ctx.lineTo(10, 50);
  ctx.lineTo(80, 120);
  ctx.stroke();
  const _img = document.createElement('img');
  _img.src = canvas.toDataURL();
  expect(canvas.toDataURL()).toMatchSnapshot();
});
