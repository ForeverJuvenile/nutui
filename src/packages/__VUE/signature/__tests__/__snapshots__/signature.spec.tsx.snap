// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`props custom-class 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAZAAAAGQCAYAAACAvzbMAAAAAXNSR0IArs4c6QAAAARzQklUCAgICHwIZIgAAAS5SURBVHic7djBTRtRFIbR30Huw0UM0mypIKWEDgIVJCklFbC1BEW4DwvIIkghicEDeGbem3fOcu5b3N2nuQkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADzpuu6y7/vN3HsAML9PQx92XXf1+Pj4bb/f34gIAKshj57i8fXZp916vb7Ybre7cdYCoHRHA3J+fv7l4eHh+4GRiAA07OgJ6+zs7GeS3YHRxjkLoF2DTlh932/2+/1Nks2BsT8RgAYNCkgiIgD8bXBAEhEB4I83BSQREQB+e3NAEhEB4J0BSUQEoHXvDkgiIgAt+1BAEhEBaNWHA5KICECLThKQREQAWnOygCQiAtCSkwYkERGAVpw8IImIALRglIAkIgKwdKMFJBERgCUbNSCJiAAs1egBSUQEYIkmCUgiIgBLM1lAEhEBWJJJA5KICMBSTB6QREQAlmCWgCQiAlC72QKSiAhAzWYNSCIiALWaPSCJiADUqIiAJCICUJtiApKICEBNigpIIiIAtSguIImIANSgyIAkIgJQumIDkogIQMmKDkgiIgClKj4giYgAlKiKgCQiAlCaagKSiAhASaoKSCIiAKWoLiCJiACUoMqAJCICMLdqA5KICMCcqg5IIiIAc6k+IImIAMxhEQFJRARgaosJSCIiAFNaVEASEQGYyuICkogIwBQWGZBERADGttiAJCICMKZFByQREYCxLD4giYgAjKGJgCQiAnBqzQQkERGAU2oqIImIAJxKcwFJRATgFJoMSCIiAB/VbEASEQH4iKYDkogIwHs1H5BERADeQ0CeiAjA2wjIMyICMJyA/ENEAIYRkANEBOA4AXmBiAC8TkBeISIALxOQI0QE4DABGUBEAP73ae4FarDdbnfr9foiye7AeHN/f/954pUAZicgA70UkdVqdX17e/tjnq0A5uOE9UbPz1mr1er67u7uau6dAKhE3/ebrusu594DAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACAUv0CXZyGEdaVWMgAAAAASUVORK5CYII="`;
